### PR TITLE
[DistDialect] Dist Interface

### PR DESCRIPTION
**Pull request by JZ-LIANG**
### PR types
New features

### PR changes
Others

### Description

Pcard-67164

add dist interface for auto parallel related method (process_mesh, dims_mapping, dist_attr, etc). 
and add local_type function as the first implement.

### CODE REVIEW

File: paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/pir/dialect/distributed/ir/dist_interface.h"
+
+namespace paddle::dialect {}  // namespace paddle::dialect
```
Comment (winter-wang) [inline paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc:17]:
```suggestion
```

File: paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (winter-wang) [inline paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc:1]:
```suggestion
// Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
```

**Review by winter-wang — APPROVED** [review]
LGTM  overall

File: paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc (hunk 2 above)
Comment (JZ-LIANG) [reply paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc:1]: fixed in next pr.

File: paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc (hunk 1 above)
Comment (JZ-LIANG) [reply paddle/fluid/pir/dialect/distributed/ir/dist_interface.cc:17]: fixed in next pr.

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**